### DISCUSSION TITLE
importing 'additionalPaths'?

### DISCUSSION THREAD

**Discussion by GabrielMahan** [ElMassimo/vite_ruby#78] [Q&A] [answered]
I'm trying to import some shared scss files stuck in my old `app/assets` dir, (my `sourceCodeDir` is `app/javascript`) 

I see in the migration guide that to reference code outside the sourceCodeDir we should add the additional paths with a glob, so in my vite.json I've added 

```
"watchAdditionalPaths": [
  "app/assets/**/*"
]
```

How do we reference these files in a vue component?  In particular in the stylesheet?

When I try something like the following in a vue component

```
<style lang="scss">
// @import '~/app/assets/stylesheets/shared.scss'; ??? 
</style> 
```
it looks that the path is still resolving to my `app/javascript` folder

**Comment by ElMassimo** [comment, marked as answer]
[sourceCodeDir]: https://vite-ruby.netlify.app/config/#sourcecodedir

Hi Gabriel!

As it's mentioned in the [_Aliases_ section of the docs](https://vite-ruby.netlify.app/guide/development.html#import-aliases-%F0%9F%91%89), the `~` prefix is an alias to your <kbd>[sourceCodeDir]</kbd>, which in your case is `app/javascript`.

You have a few options to reference an outside file:

- Using relative paths: 👎 error-prone and breaks if you move the file
- Using relative paths with the root alias as in `~/../assets`: 👎 harder to understand
- Adding an alias in `vite.config.ts` ✅

I'd recommend adding an [alias](https://vitejs.dev/config/#resolve-alias) in your `vite.config.ts`:

```js
import { resolve } from 'path'
import { defineConfig } from 'vite'

export default defineConfig({
  resolve: {
    alias: {
      '@assets': resolve(__dirname, 'app/assets'),
    },
  },
```

There is a very similar [alias being used](https://github.com/ElMassimo/vite_ruby/blob/main/vite-plugin-ruby/example/app/frontend/entrypoints/main.ts#L4) in [this example setup](https://github.com/ElMassimo/vite_ruby/blob/main/vite-plugin-ruby/example/vite.config.ts#L10).

I think this is common enough that it might be worth mentioning it in the docs. Let me know if that helps!

**Reply by ElMassimo** [reply]
Documented this in <kbd>[watchAdditionalPaths](https://vite-ruby.netlify.app/config/#watchadditionalpaths)</kbd>, and cross-linked to it from the [_Import Aliases_ section](https://vite-ruby.netlify.app/guide/development.html#import-aliases-%F0%9F%91%89).

**Reply by GabrielMahan** [reply]
@ElMassimo thank you! Your response was super clear and very helpful! You've saved me a lot of headaches!!
